### PR TITLE
HFP-3905 run validate before pack with '-f' option to skip it

### PR DESCRIPTION
**Pull request by devland**

### CODE REVIEW

**Review by otacke — COMMENTED** [review]
Just some comments for your consideration regarding the code.

I think I also found one issue, see ticket.

File: assets/utils/cli.js [hunk 1]
```diff
@@ -481,6 +481,7 @@ var commands = [
     shortDescription: 'Packs the given libraries',
     description:
       'Use -r for recursive packaging, will pack all dependencies as well' + lf +
+      'Use -f to skip library validation' + lf +
```
[the remark this answers is not in the document]
Comment (devland) [reply assets/utils/cli.js:484]: I'm not sure.
The original requirement was to have it optional but enabled by default.

File: assets/utils/cli.js [hunk 2]
```diff
@@ -494,7 +495,24 @@ var commands = [
...
+        let ok = true;
+        for (let item of result) {
+          if (item.status !== 'ok') {
+            ok = false;
+            break;
+          }
+        }
+        if (!ok) {
+          console.log('validation failed; use \'-f\' to skip validation');
+          return;
```
[the remark this answers is not in the document]
Comment (devland) [reply assets/utils/cli.js:511]: Unless this is used in general, and it's not, it makes little sense to only have it here.
It's a good idea for a future refactoring, though.

File: assets/utils/cli.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (otacke) [reply assets/utils/cli.js:484]: Alright. I'll ask Hanna then. Feels weird to leave an option to build packages that don't validate.

**Pull request merged**